### PR TITLE
misc: refine build system and tooling

### PR DESCRIPTION
**Pull request by mochaaP**
- Add editorconfig and enforce rules
- Format CMakeLists.txt with cmake-format
- Use CTest, add test target
- Install qnearbyshared to libexec
- Remove Protobuf absl workaround
- Add more sanitizer support by specifying CMAKE_BUILD_TYPE

### CODE REVIEW

File: qnearbyshare-receive/CMakeLists.txt [hunk 1]
```diff
@@ -1,19 +1,13 @@
-set(SOURCES
-        main.cpp
+set(SOURCES main.cpp)
 
-        )
-
-set(HEADERS
-
-        receiver.cpp receiver.h)
+set(HEADERS receiver.cpp receiver.h)
```
[the remark this answers is not in the document]
Comment (mochaaP) [reply qnearbyshare-receive/CMakeLists.txt:3]: done :)

**Comment by vicr123** [comment]
I'm on Arch and it seems the protobuf absl workaround is still required :/

```
FAILED: qnearbyshared/qnearbyshared 
: && /usr/bin/c++ -g  qnearbyshared/CMakeFiles/QNearbyShare-daemon.dir/QNearbyShare-daemon_autogen/mocs_compilation.cpp.o qnearbyshared/CMakeFiles/QNearbyShare-daemon.dir/main.cpp.o qnearbyshared/CMakeFiles/QNearbyShare-daemon.dir/dbus/dbusnearbysharemanager.cpp.o qnearbyshared/CMakeFiles/QNearbyShare-daemon.dir/dbus/dbusnearbysharesession.cpp.o qnearbyshared/CMakeFiles/QNearbyShare-daemon.dir/dbus/dbushelpers.cpp.o qnearbyshared/CMakeFiles/QNearbyShare-daemon.dir/dbus/dbusnearbysharelistener.cpp.o qnearbyshared/CMakeFiles/QNearbyShare-daemon.dir/dbus/dbusnearbysharediscovery.cpp.o -o qnearbyshared/qnearbyshared  libqnearbyshare-server/libqnearbyshare-server.a  dbus-types/liblibqnearbyshare-dbus-types.a  libqnearbyshare-server/QtZeroConf/libQtZeroConf.a  /usr/lib/libQt6Network.so.6.5.2  /usr/lib/libavahi-client.so  /usr/lib/libavahi-common.so  libqnearbyshare-server/proto/libqnearbyshared-proto.a  /usr/lib/libprotobuf.so  /usr/lib/libcryptopp.so  /usr/lib/libQt6DBus.so.6.5.2  /usr/lib/libQt6Core.so.6.5.2 && :
/usr/bin/ld: libqnearbyshare-server/proto/libqnearbyshared-proto.a(offline_wire_formats.pb.cc.o): undefined reference to symbol '_ZN4absl12lts_2023080212log_internal17MakeCheckOpStringIllEEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEET_T0_PKc'
/usr/bin/ld: /usr/lib/libabsl_log_internal_check_op.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

**Comment by mochaaP** [comment]
Try updating the `FindProtobuf.cmake` in your `<datadir>/cmake/Modules`, it's possibly lacking behind.

I tested this on Fedora rawhide.

**Comment by mochaaP** [comment]
Or we ship a `FindProtobuf.cmake` ourselves.

**Comment by vicr123** [comment]
The latest version of protobuf that I can get without compiling it myself seems to be 24.2. I'm open to the idea of shipping our own `FindProtobuf.cmake` as long as it works with the system provided protobuf

**Review by vicr123 — APPROVED** [review]

**Comment by vicr123** [comment]
Merged, thanks for the PR!

**Pull request merged**